packages/ckeditor5-ckbox/src/ckboxconfig.ts

```diff
@@ -137,8 +137,131 @@ export interface CKBoxConfig {
 	 * Defaults to {@link module:utils/locale~Locale#uiLanguage `Locale#uiLanguage`}
 	 */
 	language?: string;
+
+	/**
+	 * This option allows opening CKBox in dialog mode. It takes a boolean value or
+	 * a configuration object with the following attributes:
+	 */
+	dialog?: CKBoxDialog;
+
+	/**
+	 * Allows setting custom icons for categories.
+	 */
+	categories?: CKBoxCategories;
+
+	/**
+	 * Configures the view of CKBox.
+	 */
+	view?: CKBoxView;
+
+	/**
+	 * Configures when dialog should be minimized and hidden.
+	 */
+	upload?: CKBoxUpload;
+
+	/**
+	 * Specifies the file extensions considered valid for user interaction. Whith this
+	 * option developers can restrict user interaction to only those assets whose file
+	 * extensions match those listed in the array. Assets whose file
+	 * extensions are not listed in the choosableFileExtensions array are
+	 * automatically disabled within the CKBox interface.
+	 *
+	 * ```ts
+	 * const ckboxConfig = {
+	 *		choosableFileExtensions: ['jpg', 'png']
+	 * };
+	 * ```
+	 */
+	choosableFileExtensions?: Array<string>;
 }
 
+export type CKBoxDialog = {
+
+	/**
+	 * The dialog width in pixels.
+	 */
+	width: number;
+
+	/**
+	 * The dialog height in pixels.
+	 */
+	height: number;
+};
+
+export type CKBoxCategories = {
+
+	/**
+	 * This option takes an object with categories and icons that should be used instead
+	 * of the default ones. Categories can be defined using either their name or id.
+	 * Icons should be defined as strings containing the SVG images, or as React components.
+	 *
+	 * ```ts
+	 * const ckboxConfig = {
+	 * 		categories: {
+	 * 				icons: {
+	 * 					Images: '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path ... /></svg>',
+	 * 					// Category can be referenced by ID
+	 * 					// 'fdf2a647-b67f-4a6c-b692-5ba1dc1ed87b': '<svg...'
+	 * 				}
+	 * 		}
+	 * }
+	 * ```
+	 */
+	icons: {
+		[ key: string ]: string;
+	};
+};
+
+export type CKBoxView = {
```

Comment:
I'd add a suffix `Config` to this type name. Like `ToolbarConfig` or `LanguageConfig` are named. In case, we have a class that actually represents a `CKBoxDialog`. And for consistency.